layers/cmd/valueset.go

```diff
@@ -311,6 +319,222 @@ func LoadValuesetFiles(vs *Valuesets, files []string) error {
 	return nil
 }
 
+var options_re = regexp.MustCompile(`options`)
+var separator_re = regexp.MustCompile("=.")
+
+func parseSpreadsheet(sheet [][]string) (optionsRows [][]string, headers []string, headerIdx int, data [][]string, err error) {
+ROWS:
+	for rowIdx, row := range sheet {
+		if len(row) == 0 {
+			continue
+		}
+		for _, columnData := range row {
+			if options_re.MatchString(columnData) {
+				optionsRows = append(optionsRows, row)
+				continue ROWS
+			} else {
+				headers = row
+				headerIdx = rowIdx
+				break ROWS
+			}
+		}
+	}
+	data = sheet[headerIdx+1:][:]
+	if len(data) == 0 {
+		err = fmt.Errorf("Empty data")
+	}
+	return optionsRows, headers, headerIdx, data, err
+}
+
+func parseOptions(optionsRows [][]string) map[string][]string {
+	optionsMap := make(map[string][]string, len(optionsRows))
+	for _, opt := range optionsRows {
+		optionType := opt[0]
+		// options.lookupOrder -- code | text | ...
+		optionsMap[optionType] = opt[1:]
+	}
+	return optionsMap
+}
+
+func parseData(vsets *Valuesets, sheetName string, headers []string, data [][]string, optionsMap map[string][]string, headerRowStart int) {
+	// lookupOrder (header based search)
+	if len(optionsMap) > 0 {
+		if lookup_opt, exists := optionsMap["lookupOrder"]; exists {
+			var headers_with_sep map[string]struct{}
+			var sep string
+		HEADERS_OPTIONS:
+			for hdrIdx, hdr := range headers {
+				if hdr == lookup_opt[hdrIdx] {
+					if sep_opt, exists := optionsMap["separator"]; exists {
+						// assume all separators are the same for each header
+						// =; or =,
+						sep = separator_re.FindString(sep_opt[0])
```

Comment:
Use strings.Split instead of a RE, and trim each element. Make sure empty separator means space. Deal with this when parsing options